execution_engine_testing/tests/src/test/system_contracts/auction/mod.rs

```diff
@@ -171,26 +201,73 @@ fn should_support_contract_staking() {
             .query(None, delegation_key, &[])
             .expect("should have delegation bid")
     {
-        // assert_ne!(
-        //     delegator.staked_amount(),
-        //     delegate_amount,
-        //     "staked amount should execeed delegation amount due to rewards"
-        // );
         delegator.staked_amount()
     } else {
         U512::zero()
     };
 
-    // unstake from contract
+    // restake from contract
     builder
         .exec(
             ExecuteRequestBuilder::contract_call_by_name(
                 account,
                 &contract_name,
-                &unstake,
+                &entry_point_name,
                 runtime_args! {
+                    ARG_ACTION => restake,
                     ARG_AMOUNT => increased_delegate_amount,
                     ARG_VALIDATOR => validator_pk.clone(),
+                    ARG_NEW_VALIDATOR => other_validator_pk.clone()
+                },
+            )
+            .build(),
+        )
+        .commit()
+        .expect_success();
+
+    assert!(
+        builder.query(None, delegation_key, &[]).is_err(),
+        "delegation record should be removed"
+    );
+
+    assert_eq!(
+        post_delegation_balance,
+        builder.get_purse_balance(contract_purse),
+        "at this point, unstaked token has not been returned"
+    );
+
+    for _ in 0..=7 {
```

Comment:
It used to be the old auction delay, so I think we spots in our code where we use the naked u8 instead of the old constant

Reply:
It's actually the `unbonding delay` rather than the auction delay, which is relevant because I'm unbonding and rebonding at various points. I cheated and hardcoded the typical value, but @zajko is correct that it would be more correct to use the configured value. The builder does offer a getter for it already, so it is trivial to switch to it and I will do so.